src/javis_viewer.jl

```diff
@@ -342,20 +343,37 @@ Internal method for livestreaming
 """
 function _livestream(
     streamconfig::StreamConfig,
-    framerate::Int,
-    width::Int,
-    height::Int,
-    pathname::String,
+    framerate,
+    video,
+    objects,
+    layers,
+    frames,
+    tempdirectory,
 )
+    # kill any existing stream
     cancel_stream()
 
     livestreamto = streamconfig.livestreamto
+    config_frames = streamconfig.frames
     twitch_key = streamconfig.twitch_key
 
     if livestreamto == :twitch && isempty(twitch_key)
         return error("Please enter your twitch stream key")
     end
 
+    stream_frames = config_frames == :all ? frames : config_frames
+    stream_filecounter = 1
+    for frame in stream_frames
+        frame_image = convert.(RGB, get_javis_frame(video, objects, frame; layers = layers))
+        if !isempty(tempdirectory)
```

Comment:
what happens if tempdirectory is empty? It just generates the frames but can't use it?

Reply:
No tempdir is created before if it's empty (as before)
https://github.com/Wikunia/Javis.jl/pull/387/files/56156371a5657544c632a3cb6faa2b39681e1fe7#diff-763ffddd0fe1ffb533564007035f5947812905676655f155ce1ee4357eb53c5eR275

Reply:
Yeah what I mean is: it should never be empty so we don't need the if. The problem is if we have an mp4 as the pathname without a temp directory but want to stream it will fail, right?